api/test/unit/views/media_views_test.py

```diff
@@ -67,6 +70,27 @@ def send(session, req, **kwargs):
     return fixture
 
 
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    "media_type, media_factory, viewset",
+    [
+        ("images", ImageFactory, ImageViewSet),
+        ("audio", AudioFactory, AudioViewSet),
+    ],
+)
+def test_retrieve_query_count(api_client, media_type, media_factory, viewset, requests):
+    media = media_factory.create()
+    factory = APIRequestFactory()
+    request = factory.get(f"/v1/{media_type}/{media.identifier}/")
+
+    # This number goes up without `select_related` in the viewset queryset.
+    with pytest_django.asserts.assertNumQueries(1):
+        view = viewset.as_view({"get": "retrieve"})
+        res = view(request, identifier=media.identifier)
```

Comment:
Totally optional, but is it possible to simplify this using the `api_client` fixture? It actually exercises the entire route including middleware, not just the view itself, so is slightly more comprehensive. It also allows the user to avoid having to instantiate the view and call it themselves like we've done here.

Reply:
IIRC I tried that first but that makes a request to the endpoint from outside so `assertNumQueries` always sees 0 queries.